src/render/draw_raster.js

```diff
@@ -122,7 +122,7 @@ function getFadeValues(tile, parentTile, layer, transform) {
         // if no parent or parent is older, fade in; if parent is younger, fade out
         const fadeIn = !parentTile || Math.abs(parentTile.coord.z - idealZ) > Math.abs(tile.coord.z - idealZ);
 
-        const childOpacity = util.clamp(fadeIn ? sinceTile : 1 - sinceParent, 0, 1);
+        const childOpacity = (fadeIn && tile.refreshedUponExpiration) ? 1 : util.clamp(fadeIn ? sinceTile : 1 - sinceParent, 0, 1);
```

Comment:
@anandthakker good question. TBH I can't tell you exactly how the tile fading logic works, _but_ you'd only want to bypass fading logic if a tile is fading in (because it fades in newly loaded tiles). That…may not be a good enough explanation. 🤔